### PR TITLE
Fix crash in JNI getApproximateSizes

### PR DESCRIPTION
**Pull request by koldat**
This change is fixing a crash happening in getApproximateSizes JNI implementation. It also reenables Java test that was crashing most likelly because if this bug.

### CODE REVIEW

**Comment by koldat** [comment]
@adamretter , please take a look

File: java/rocksjni/rocksjni.cc [hunk 1]
```diff
@@ -2277,10 +2277,11 @@ jlongArray Java_org_rocksdb_RocksDB_getApproximateSizes(
 
   auto ranges = std::unique_ptr<ROCKSDB_NAMESPACE::Range[]>(
       new ROCKSDB_NAMESPACE::Range[range_count]);
+  size_t range_offset = 0;
   for (jsize i = 0; i < jlen; ++i) {
     auto* start = reinterpret_cast<ROCKSDB_NAMESPACE::Slice*>(jranges[i]);
     auto* limit = reinterpret_cast<ROCKSDB_NAMESPACE::Slice*>(jranges[++i]);
-    ranges.get()[i] = ROCKSDB_NAMESPACE::Range(*start, *limit);
+    ranges.get()[range_offset++] = ROCKSDB_NAMESPACE::Range(*start, *limit);
```
Comment (adamretter) [inline java/rocksjni/rocksjni.cc:2284]: So I think that on x64 (macOS):

1. `size_t` is `uint64_t`
2. `jsize` is `int`, which is 32 bits.

So I don't quite understand the change for this issue.... As previously `i` could only be 32 bits, why does increasing it to 64 bits fix this, when the max could only ever be 32 bits.

File: java/src/test/java/org/rocksdb/RocksDBTest.java [hunk 2]
```diff
@@ -1185,7 +1184,7 @@ public void getApproximateSizes() throws RocksDBException {
 
         final long[] sizes = db.getApproximateSizes(
             Arrays.asList(
-                new Range(new Slice(key1), new Slice(key2)),
+                new Range(new Slice(key1), new Slice(key1)),
```
Comment (adamretter) [inline java/src/test/java/org/rocksdb/RocksDBTest.java:1187]: Why the change in the range, is it a genuine mistake originally?

File: java/rocksjni/rocksjni.cc (hunk 1 above)
Comment (koldat) [reply java/rocksjni/rocksjni.cc:2284]: Issue is that i variable is increasing two times faster thus code writes after allocated memory.

File: java/src/test/java/org/rocksdb/RocksDBTest.java (hunk 2 above)
Comment (koldat) [reply java/src/test/java/org/rocksdb/RocksDBTest.java:1187]: Because test is asserting against 0 size. It cannot happen it is zero size when it gets range that is non empty.

File: java/rocksjni/rocksjni.cc (hunk 1 above)
Comment (adamretter) [reply java/rocksjni/rocksjni.cc:2284]: Doh! Of course!

File: java/src/test/java/org/rocksdb/RocksDBTest.java (hunk 2 above)
Comment (adamretter) [reply java/src/test/java/org/rocksdb/RocksDBTest.java:1187]: Okie dokie :-)

**Comment by adamretter** [comment]
@pdillinger LGTM! Can we merge?
